Sources/NonEmpty/NonEmpty+Dictionary.swift

```diff
@@ -106,8 +106,8 @@ extension NonEmpty where C: _DictionaryProtocol {
 
 extension NonEmpty where C: _DictionaryProtocol, C.Value: Equatable {
   public static func == (lhs: NonEmpty, rhs: NonEmpty) -> Bool {
-    return Dictionary(uniqueKeysWithValues: Array(lhs))
-      == Dictionary(uniqueKeysWithValues: Array(rhs))
+    return Dictionary<C.Key, C.Value>(uniqueKeysWithValues: Array<Element>(lhs))
```

Comment:
This now fails to compile. The type-checker can no longer infer the generics. Bug opened here: https://bugs.swift.org/browse/SR-9663

Reply:
Seems like swift can't properly infer types when there are labeled tuples involved. I've fixed it like this:

```swift
extension NonEmpty where C: _DictionaryProtocol, C.Value: Equatable {
  public static func == (lhs: NonEmpty, rhs: NonEmpty) -> Bool {
    let lhs = Array(lhs)
    let rhs = Array(rhs)
    return Dictionary(uniqueKeysWithValues: lhs)
      == Dictionary(uniqueKeysWithValues: rhs)
  }
}
```

https://github.com/ilyapuchka/swift-nonempty/commit/c82e104312a2539b409594faee31bc1effe67eb0#diff-b4f6b5a4e463366aef93a74e1d7a0c5c

Reply:
@ilyapuchka Are you still seeing this problem? The core team seems to have fixed the regression.

Reply:
I still have last year snapshot of Swift 5, so I bet this was fixed later, but still wanted to share.